src/vm/i386/cgenx86.cpp

```diff
@@ -299,7 +299,12 @@ void TransitionFrame::UpdateRegDisplay(const PREGDISPLAY pRD)
 
     MethodDesc * pFunc = GetFunction();
     _ASSERTE(pFunc != NULL);
+
+#ifdef WIN64EXCEPTIONS
+    UpdateRegDisplayHelper(pRD, 0);
```

Comment:
@parjong I have spent some time today thinking about all of the aspects of the problem. Here are my notes:
1. We cannot store the esp at the call instruction in the context, unless we change the UnwindStackFrame to work with that semantics of esp directly. I don't see a way how OOPStackUnwinderX86::VirtualUnwind could augment it back to the SP before pushing the stack args. The problem is that the unwind gets the current context, but for getting the size of the stack arguments, we would need the previous context that we've unwound to the current context before. And that's not available at that point. 
2. There are actually two different ways the stack is unwound. 
  a. One is in the stackwalk.cpp, where it doesn't directly walk native code frames, but it uses the explicit stack frames (classes derived from the Frame class) to cross the native area. That's where the XXXFrame::UpdateRegDisplay is used. Frames derived from the TransitionFrame contain MethodDesc of the function that gets called after the transition. There is a managed method A that calls a native code with the TransitionFrame and then a managed method B is called. But logically, you can see it as method A calling method B. And so the stack arguments size from the MethodDesc of method B stored in the transition frame gives you the delta between the caller SP stored in the regdisplay passed to the UpdateRegDisplay and the SP where the stack arguments were pushed. So the TransitionFrame::UpdateRegDisplay should always call the UpdateRegDisplayHelper with the pFunc->CbStackPop(), even for WIN64EXCEPTIONS in the current model when the context of managed frames contains esp before the stack arguments were pushed and not the caller SP. First for the sake of consistency and second  because calling the EECodeManager::GetCallerSp on REGDISPLAY of the method A would otherwise return nonsense.
  b. The other one is in the UnwindManagedExceptionPassX and Thread::VirtualUnwindToFirstManagedCallFrame where we walk every single native frame. In this case, the PAL_VirtualUnwind is used and so when we walk out of native frames to a managed frame, the esp is obviously the caller SP and not the SP before the stack arguments were pushed. So at this place we would need to have a way to lookup the size of stack arguments from an address in the function. It seems we could use the information in the jithelpers.h to build some kind of lookup.